src/App.js

```diff
@@ -1,16 +1,38 @@
-import React from 'react';
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [data, setData] = useState(chatMessages);
+
+  const updateEntries = (updatedEntry) => {
+    const entries = data.map((entry) => {
+      const returnEntry = entry.id === updatedEntry.id ? updatedEntry: entry;
+      return returnEntry;
+    });
+    setData(entries);
+  };
```

Comment:
Nice work updating the state data by returning a brand new object with `map` and the spread operator!

Here's a refresher on updating state in React:

React components re-render whenever there is a change/update to state data. In this case, we are modifying a key-value pair within an object in a nested data structure.

Under the hood, React uses `Object.is()` to compare the previous state object with the one that's been provided via`setMessages(updatedMessages`). `Object.is()` checks if the object passed into the set state function has a different reference in memory. In Javascript, changing the properties and/or values in an object does NOT change the object reference, which is why we must create a new version of our state object (contains a copy of all the properties that weren't updated along with the properties/values that were).

Here is an article with more info: https://www.valentinog.com/blog/react-object-is/

Reply:
We can further refactor by removing the assignment of `returnEntry`. The ternary in`map` decides which object to append to a list and `map` will return that list and assign it to `entries`. 

```javascript
  const updateEntries = (updatedEntry) => {
    const entries = data.map((entry) => {
      entry.id === updatedEntry.id ? updatedEntry: entry;
    });
    setData(entries);
  };
```